content/blog/apl/2020-07-11_repository_identification.md

```diff
@@ -0,0 +1,99 @@
+
+```
+https://eu.hpkg.haiku-os.org/haikuports/master/x86_64/current
+```
+
+The repository also contains a special file `repo.info` that describes the repository found at a given `baseUrl`.  Here is an example of what this might look like;
+
+```
+name HaikuPorts
+vendor "Haiku Project"
+summary "The HaikuPorts repository"
+priority 1
+identifier e90b7d69-969c-4c26-8894-06f965818c7d
+architecture x86_64
+```
+
+The repository also contains a file with a well known filename `repo`.  This `repo` file is in an optimized Haiku-specific [format](https://github.com/haiku/haiku/blob/master/docs/develop/packages/FileFormat.rst) called "HPKR" and it contains a catalog of all of the packages that are available in the repository as well as some meta-data about the files such as their copyrights, English-language summaries etc...
+
+Finally the repository also contains the packages themselves.  These files are in a directory `packages` with a leaf-name that includes the package name as well as the version and target computing architecture.  For example;
+
+* `armyknife-5.1.0-4-x86_64.hpkg`
+* `artpaint-2.1.0git-7-x86_64.hpkg`
+
+These package files are conveyed in a different optimized Haiku-specific format called "HPKG".  This typically contains binaries to be executed on a Haiku system as well as resources such as images and documentation to support the software included in the package.
+
+## Where Do The Repository Data Files Go?
+
+When a user registers a repository onto their Haiku system, the system will download the `repo.info` file and the `repo` HPKR file.  By means of example, considerating the configuration of the HaikuPorts repository, the HPKR file is stored at...
+
+```
+/system/cache/package-repositories/HaikuPorts
+```
+
+...and the `repo.info` lands at...
+
+```
+/system/settings/package-repositories/HaikuPorts
+```
+
+## Mirrors
+
+A mirror is a copy of a repository at a different `baseUrl`.  There are a number of reasons why a mirror is a good idea.
+
+* A single server and it's bandwidth may be exhausted by a large number of users accessing it at once.
+* If a single server fails then it may impact a large number of users of the platform and a backup location would provide an alternative.
+* Users may be in very different geographies.  For example, Germany and New Zealand are on the opposite sides of the planet.  If a significant number of users from New Zealand access the repository in Germany then this would mean a potentially large volume of international internet traffic would unnecessarily traverse the globe.  It is better if the traffic were contained in the Oceania region of the world where New Zealand were located because this would put less strain on international links and would also be faster for the New Zealand users.
```

Comment:
```suggestion
* Users may be in very different geographies.  For example, Germany and New Zealand are on the opposite sides of the planet.  If a significant number of users from New Zealand access the repository in Germany then this would mean a potentially large volume of international internet traffic would unnecessarily traverse the globe.  It is better if the traffic were contained in the Oceania region of the world where New Zealand is located, because this would put less strain on international links and would also be faster for the New Zealand users.
```